petapm.h

```diff
@@ -22,6 +22,12 @@ typedef struct {
     void (*readout)(int i, double * mesh, double weight);
 } PetaPMFunctions;
 
+typedef struct {
+    void (*global_readout)(int64_t k2, int kpos[3], pfft_complex * value);
+    void (*global_analysis)(void);
+    void (*global_transfer)(int64_t k2, int kpos[3], pfft_complex * value);
```

Comment:
May want to add arguments for the analysis result and parameter of the transfer.

Reply:
void * pointers to preserve the generic algorithm property?

Reply:
There is no-where for them to go - let me leave it as it is. It sets some structures which are effectively private members of the gravpm object. The other way I would have to pass the structures from petapm back to gravpm which would be a bit messy.

Reply:
Sure. when we start to do the next round of global variables cleanups ...

Reply:
A never-ending task